runtime-modules/content/src/vnft_auction/types.rs

```diff
@@ -0,0 +1,279 @@
+        Ok(())
+    }
+}
+
+/// Owned vNFT representation
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct OwnedNFT<
+    AccountId,
+    Moment: BaseArithmetic + Copy,
+    CuratorGroupId: Default + Copy,
+    CuratorId: Default + Copy,
+    MemberId: Default + Copy,
+    Balance,
+> {
+    pub owner: AccountId,
+    pub transactional_status:
+        TransactionalStatus<AccountId, Moment, CuratorGroupId, CuratorId, MemberId, Balance>,
+    pub creator_royalty: Option<(AccountId, Royalty)>,
+}
+
+impl<
+        AccountId: PartialEq,
+        Moment: BaseArithmetic + Copy,
+        CuratorGroupId: Default + Copy,
+        CuratorId: Default + Copy,
+        MemberId: Default + Copy,
+        Balance,
+    > OwnedNFT<AccountId, Moment, CuratorGroupId, CuratorId, MemberId, Balance>
+{
+    pub fn is_owner(&self, account_id: &AccountId) -> bool {
+        self.owner.eq(account_id)
+    }
+}
+
+/// Enum, representing nft issuance status
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum NFTStatus<
+    AccountId,
+    Moment: BaseArithmetic + Copy,
+    CuratorGroupId: Default + Copy,
+    CuratorId: Default + Copy,
+    MemberId: Default + Copy,
+    Balance,
+> {
+    NoneIssued,
+    Owned(OwnedNFT<AccountId, Moment, CuratorGroupId, CuratorId, MemberId, Balance>),
+}
+
+impl<
+        AccountId,
+        Moment: BaseArithmetic + Copy,
+        CuratorGroupId: Default + Copy,
+        CuratorId: Default + Copy,
+        MemberId: Default + Copy,
+        Balance,
+    > Default for NFTStatus<AccountId, Moment, CuratorGroupId, CuratorId, MemberId, Balance>
+{
+    fn default() -> Self {
+        Self::NoneIssued
+    }
+}
+
+/// Either new auction, which requires vNFT issance or auction for already existing nft.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum AuctionMode {
+    // Auction, where nft issued at the end
+    WithIssuance(Option<Royalty>, Metadata),
+    // Auction for already existing nft
+    WithoutIsuance,
+}
+
+impl Default for AuctionMode {
+    fn default() -> Self {
+        Self::WithoutIsuance
+    }
+}
+
+/// Information on the auction being created.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct AuctionRecord<
+    AccountId,
+    Moment: BaseArithmetic + Copy,
+    CuratorGroupId: Default + Copy,
+    CuratorId: Default + Copy,
+    MemberId: Default + Copy,
+    Balance,
+> {
+    pub auctioneer: ContentActor<CuratorGroupId, CuratorId, MemberId>,
+    pub auctioneer_account_id: AccountId,
+    pub auction_mode: AuctionMode,
+    pub starting_price: Balance,
+    pub buy_now_price: Option<Balance>,
+    pub round_time: Moment,
```

Comment:
- I assume this is the duration of the auction? if so, could we rename it to `auction_duration`.
- Remember that we also need to be able to issue an NFT which you only can start bidding on at some block in the future, from spec `which starts at some predefined time in the future`.

Reply:
> * I assume this is the duration of the auction? if so, could we rename it to `auction_duration`.
> * Remember that we also need to be able to issue an NFT which you only can start bidding on at some block in the future, from spec `which starts at some predefined time in the future`.

Should there be a necessity to specify both block number and timestamp for user or just a block number?

Reply:
Lets stick with block number only, that is the only thing we do in the runtime so far, so lets keep that for consistently.

Reply:
Done https://github.com/Joystream/joystream/pull/2512/commits/0c3c5910954a2d0bf438bb8850aa577ab1183c7d